src/main/java/com/avispl/symphony/dal/device/axis/m3064/AxisCommunicator.java

```diff
@@ -402,11 +417,11 @@ private int getNoOfHistoricalMetric() {
 	 */
 	private AdvancedControllableProperty controlSwitch(Map<String, String> control, Map<String, String> stats, AxisControllingMetric axisControllingMetric) {
 		stats.put(axisControllingMetric.getName(), control.get(axisControllingMetric.getName()));
-		if (!control.get(axisControllingMetric.getName()).equals(AxisConstant.NONE)) {
-			return control.get(axisControllingMetric.getName()).equals(AxisConstant.YES) ? createSwitch(axisControllingMetric.getName(), 1)
-					: createSwitch(axisControllingMetric.getName(), 0);
+		if (!control.get(axisControllingMetric.getName()).equals(AxisConstant.NONE)
```

Comment:
Do you need to have 2 checks? If it's gonna be equal to "YES", wouldn't this mean, that it's not equal to "NONE"? :)